### ISSUE TITLE
[Bug]:  (just a reminder) Stretchly: Unity not supported for DND detection, yet.

### ISSUE THREAD

**Issue by eduarddrenth** [hovancik/stretchly#1528]
### Version

- [X] I'm using version 1.17

### Known issues

- [X] I've checked Known issues

### Existing issues

- [X] I've checked Existing issues

### Advanced Preferences

- [X] I've checked Advanced Preferences

### What operating system are you using?

Linux

### Operating System Version

ubuntu 24.04.1

### Reproduction steps

1. start `journalctl -f`
2. start stretchly
3. You'll see dnd not supported in the logging and stretchly ignores dnd

### Expected Behavior

- stretchly successfully detects dnd and does not ignore it

### Actual Behavior

see reporoduction steps

### Additional information

after successfully fixing #1466 (thanks!) dnd does not work anymore on the default ubuntu-gnome desktop

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by hovancik** [comment]
So Gnome and Unity are the same or different? I am confused. 

For Gnome I am checking `gsettings get org.gnome.desktop.notifications show-banners`. I am getting DE value from `XDG_CURRENT_DESKTOP`

Is this different DE than you had in https://github.com/hovancik/stretchly/issues/1466#issuecomment-2554672752? `XDG_CURRENT_DESKTOP=ubuntu:GNOME`

**Comment by eduarddrenth** [comment]
ubuntu uses an adapted and extended version of gnome on wayland by default.

unity used to be the default but has been abandoned for quite some time now.

I do not understand why on my ubuntu 24.04 stretchly mentions unity not being supported. On what basis does stretchly concludes it is running on a unity desktop?

**Comment by eduarddrenth** [comment]
My `XDG_CURRENT_DESKTOP` is still `ubuntu:GNOME`

**Comment by eduarddrenth** [comment]
```
$ gsettings get org.gnome.desktop.notifications show-banners
true
$ gsettings get org.gnome.desktop.notifications show-banners
false
```

**Comment by eduarddrenth** [comment]
If you change this line https://github.com/hovancik/stretchly/blob/6dd2bc35026eb91e305c2dd7cc8a35dde74d41f4/app/utils/dndManager.js#L76

into

`          if (stdout === 'false' || stdout.replace(/[^0-9a-zA-Z]/g, '') === 'false') {`

It should be working on gnome I guess

**Comment by eduarddrenth** [comment]
same for mate and cinnamon, I'll test gnome

**Comment by eduarddrenth** [comment]
When debugging I see that `process.env.XDG_CURRENT_DESKTOP` contains `Unity` as opposed to `ubuntu:GNOME` in a shell environment. Could be an ubuntu bug I guess.

Both `process.env.DESKTOP_SESSION` and `process.env.XDG_SESSION_DESKTOP` contain `ubuntu`.

When I add unity to the gnome case, dnd works:
```
      case de.includes('gnome') || de.includes('unity'):
```
No need to add `stdout === 'false'` like I suggested earlier.

**Comment by hovancik** [comment]
Thanks for checking it out! I will create some virtual machine with Gnome/Ubuntu and do some tests. Really strange the values are different... I guess I will need to find a way to get values that user sees when checking for them command line?

**Comment by hovancik** [comment]
do you autostart stretchly? can you try starting stretchly after de is loaded and check again what's in ENV/dtretchly?

**Comment by eduarddrenth** [comment]
Also when starting after DE is loaded the values for XDG_CURRENT_DESKTOP are different in bash and from npm process.

Found this https://github.com/electron/electron/issues/40795

**Comment by eduarddrenth** [comment]
electron changes XDG_CURRENT_DESKTOP, see https://github.com/electron/electron/blob/ab2a4fd836d539194bc5cde5f0d665eddeb6a134/lib/browser/init.ts#L179

**Comment by hovancik** [comment]
Nice catch! We should then replace all to ORIGINAL_XDG_CURRENT_DESKTOP

**Comment by hovancik** [comment]
@eduarddrenth do you wanna make a PR? You basically solved it :)

**Comment by eduarddrenth** [comment]
fine

**Comment by hovancik** [comment]
Argh, I broke it again, my bad. Fix is here #1534 but I will take some time to release new version. Sorry!

**Comment by eduarddrenth** [comment]
Happens to the best of us. Would be nice if you could, like in Java, use
constants or enums, so you cannot make mistakes like that.

Op do 26 dec. 2024 21:33 schreef Jan Hovancik ***@***.***>:
